activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb

```diff
@@ -297,20 +298,29 @@ def initialize(pool, frequency)
         @@mutex = Mutex.new
         @@pools = {}
```

Comment:
Is not enough using `@mutex` and `@pools`?
Looks like the `@@mutex` and `@@pools` are not touched from any Reaper instance.